pkg/authorization/workspace_content_authorizer.go

```diff
@@ -130,15 +126,20 @@ func (a *workspaceContentAuthorizer) Authorize(ctx context.Context, attr authori
 	}
 
 	switch {
-	case !isUser && !isServiceAccountFromCluster:
-		// service accounts from other workspaces cannot access
+	case isServiceAccount && isForeign:
+		// Service accounts from other workspaces might conflict with local service accounts by name.
+		// Use another reason string to make this very common case clearer.
```

Comment:
@mjudeikis wondering whether we need all this logic here at all? Isn't the rule resolver doing this work already?